api/client/client_test.go

```diff
@@ -574,3 +574,47 @@ func TestGetResources(t *testing.T) {
 		})
 	}
 }
+
+type mockAccessRequestServer struct {
+	*mockServer
+}
+
+func (g *mockAccessRequestServer) GetAccessRequests(ctx context.Context, f *types.AccessRequestFilter) (*proto.AccessRequests, error) {
+	req, err := types.NewAccessRequest("foo", "bob", "admin")
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &proto.AccessRequests{
+		AccessRequests: []*types.AccessRequestV3{req.(*types.AccessRequestV3)},
+	}, nil
+}
+
+// TestAccessRequestDowngrade tests that the client will downgrade to the non stream API for fetching access requests
+// if the stream API is not available.
+func TestAccessRequestDowngrade(t *testing.T) {
+	ctx := context.Background()
+	l, err := net.Listen("tcp", "")
+	require.NoError(t, err)
+
+	m := &mockAccessRequestServer{
+		&mockServer{
+			addr:                           l.Addr().String(),
+			grpc:                           grpc.NewServer(),
+			UnimplementedAuthServiceServer: &proto.UnimplementedAuthServiceServer{},
+		},
+	}
+	proto.RegisterAuthServiceServer(m.grpc, m)
+	t.Cleanup(m.grpc.Stop)
+
+	go func() {
+		require.NoError(t, m.grpc.Serve(l))
```

Comment:
```suggestion
		assert.NoError(t, m.grpc.Serve(l))
```

Avoid "t.Fatal"-ing in goroutines.